delta/delta.go

```diff
@@ -15,12 +14,16 @@ type Delta struct {
 }
 
 // Op is the smallest "operation"
-// TODO: Handle embeds
 type Op struct {
-	Insert     []rune                 `json:"insert,omitempty"`
-	Retain     *int                   `json:"retain,omitempty"`
-	Attributes map[string]interface{} `json:"attributes,omitempty"`
-	Delete     *int                   `json:"delete,omitempty"`
+	Insert []rune `json:"insert,omitempty"`
+	// Even though the JS delta library supports any non-string type
+	// as embeds, if we look at the documentation of Quill.js, embed
+	// is described as an object with only one key. Hence we are using
+	// map to describe embeds here.
+	InsertEmbed map[string]interface{} `json:"-"`
```

Comment:
if this can only have one key, I think it may be better to use a dedicated struct for it. My reasoning is, a map tells users that you can insert many keys: values in here, but that isn't true. So maybe

```
type Embed struct{
    Key string,
    Value interface{},
}
```